distributed/s3.py

```diff
@@ -26,148 +32,635 @@
 
 get_s3_lock = threading.Lock()
 
+def split_path(path):
+    path = path.lstrip('s3://')
```

Comment:
This isn't quite what you want:

```python
In [1]: 's3://sally'.lstrip('s3://')
Out[1]: 'ally'
```

I suggest using `startswith` and slicing.